server/model/monitor.js

```diff
@@ -1615,6 +1616,25 @@ class Monitor extends BeanModel {
         return oAuthAccessToken;
     }
 
+    /**
+     * Evaluates the provided value with the expected value based on the defined operator. Fallback is the equals operator
+     * @param {any} value The expected_value that the matching operator will evaluate
+     * @returns {boolean} Is the JsonPath matching the expected value with the defined operator?
+     */
+    isJsonPathMatching(value) {
+        const operators = {
+            "==": (value) => value === this.expectedValue,
+            "!=": (value) => value !== this.expectedValue,
+            "<": (value) => value < this.expectedValue,
+            "<=": (value) => value <= this.expectedValue,
+            ">": (value) => value > this.expectedValue,
+            ">=": (value) => value >= this.expectedValue,
```

Comment:
You're right that the comparison between String & String does not make sense when numbers are included. However, string is the format to hold both, expression result & expected value.

For <, <=, >, >= casting both values to `Number` would work. Let me know if that is acceptable for you or please provide a suggestion.

Regarding user communication - I think when the compare functions take care of evaluating numbers correctly, there is no need to explain this to the user as the numbers would be casted before. If I'm missing something here, please provide an example where this would be confusing in your opinion.

Reply:
- A good approach might be to rely on js-like-type conversion.
If both values can be cast to an number without an error, cast them to a number otherwise compare them as strings and add a warning.
- An alternative would be to indiscriminately cast and add the appropriate error message.
- A third alternative would be to split these into a lexicografical and natural sorting variants to avoid confusion.
- A fourth alternative would be to only use natural sorting and add a helptext that we do compare via [`Natural sort order`](https://en.wikipedia.org/wiki/Natural_sort_order) if a relevant comparison mode is selected

In all but the 3rd case, this needs to be noted in the dropdowns helptext. (this is not clear in any case. Think about a regular user coming across this feature: do they know how this comparison works?)